File: ambeth/Ambeth.IoC/ambeth/converter/StringToClassArrayConverter.cs
using De.Osthus.Ambeth.Ioc.Annotation;
using De.Osthus.Ambeth.Log;
using De.Osthus.Ambeth.Util;
using System;
using System.Text;

namespace De.Osthus.Ambeth.Converter
{
    public class StringToClassArrayConverter : IDedicatedConverter
    {
        [LogInstance]
        public ILogger Log { private get; set; }

        [Autowired]
        public IConversionHelper ConversionHelper { protected get; set; }

        public Object ConvertValueToType(Type expectedType, Type sourceType, Object value, Object additionalInformation)
        {
            if (typeof(Type[]).Equals(expectedType))
            {
                String[] split = StringToPatternConverter.splitPattern.Split((String)value);
                Type[] result = new Type[split.Length];
                for (int a = split.Length; a-- > 0; )
                {
                    result[a] = ConversionHelper.ConvertValueToType<Type>(split[a]);
                }
                return result;
            }
            StringBuilder sb = new StringBuilder();
            Type[] array = (Type[])value;
            foreach (Type item in array)
            {
                if (sb.Length > 0)
                {
                    sb.Append(StringToPatternConverter.splitPattern.ToString());
                }
                sb.Append(item.FullName);
            }
            return sb.ToString();
        }
    }
}

File: jambeth/jambeth-ioc/src/main/java/de/osthus/ambeth/converter/StringToClassArrayConverter.java
package de.osthus.ambeth.converter;

import de.osthus.ambeth.ioc.annotation.Autowired;
import de.osthus.ambeth.util.IConversionHelper;
import de.osthus.ambeth.util.IDedicatedConverter;

public class StringToClassArrayConverter implements IDedicatedConverter
{
	@Autowired
	protected IConversionHelper conversionHelper;

	@Override
	public Object convertValueToType(Class<?> expectedType, Class<?> sourceType, Object value, Object additionalInformation)
	{
		if (Class[].class.equals(expectedType))
		{
			String[] split = StringToPatternConverter.splitPattern.split((String) value);
			Class<?>[] result = new Class<?>[split.length];
			for (int a = split.length; a-- > 0;)
			{
				result[a] = conversionHelper.convertValueToType(Class.class, split[a]);
			}
			return result;
		}
		StringBuilder sb = new StringBuilder();
		Class<?>[] array = (Class<?>[]) value;
		for (Class<?> item : array)
		{
			if (sb.length() > 0)
			{
				sb.append(StringToPatternConverter.splitPattern.pattern());
			}
			sb.append(item.getName());
		}
		return sb.toString();
	}
}
